jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorListener.kt

```diff
@@ -26,7 +27,10 @@ class CodeWhispererEditorListener : EditorFactoryListener {
                     editor.project?.let { project ->
                         PsiDocumentManager.getInstance(project).getPsiFile(editor.document)?.toProgrammingLanguage() ?. let { languageName ->
                             val language = ProgrammingLanguage(languageName).toCodeWhispererLanguage()
-                            CodeWhispererCodeCoverageTracker.getInstance(language).documentChanged(event)
+                            CodeWhispererCodeCoverageTracker.getInstance(language).apply {
+                                if (!isActive && isTelemetryEnabled()) activateTracker()
```

Comment:
would prefer this to be called `activateTrackerIfNotActive`

and move this logic there

Reply:
updated!